cli/sql.go

```diff
@@ -197,24 +213,30 @@ func runInteractive(conn *sqlConn) (exitErr error) {
 		// statement.
 		fullStmt := strings.Join(stmt, "\n")
 
-		// We save the history between each statement, This enables
-		// reusing history in another SQL shell without closing the
-		// current shell.
-		//
-		// AddHistory will push command into memory and try to persist
-		// to disk (if readline.SetHistoryPath was called).
-		// err can be not nil only if it got a IO error while
-		// trying to persist.
-		if err := readline.AddHistory(strings.Join(stmt, " ")); err != nil {
-			log.Warningf("cannot save command-line history: %s", err)
-			log.Info("command-line history will not be saved in this session")
-			readline.SetHistoryPath("")
+		if isInteractive {
+			// We save the history between each statement, This enables
+			// reusing history in another SQL shell without closing the
+			// current shell.
+			//
+			// AddHistory will push command into memory and try to persist
+			// to disk (if readline.SetHistoryPath was called).
+			// err can be not nil only if it got a IO error while
+			// trying to persist.
+			if err := readline.AddHistory(strings.Join(stmt, " ")); err != nil {
+				log.Warningf("cannot save command-line history: %s", err)
+				log.Info("command-line history will not be saved in this session")
+				readline.SetHistoryPath("")
+			}
 		}
 
 		if exitErr = runPrettyQuery(conn, os.Stdout, makeQuery(fullStmt)); exitErr != nil {
 			fmt.Fprintln(osStderr, exitErr)
 		}
 
+		if !isInteractive && exitErr != nil {
+			break
```

Comment:
So this ends processing when an error is encountered? Hmm, I suppose that's safe though it does mean I need to change my example (stupid reasons). Fine by me, though `psql` definitely just keeps going so maybe so should we.